source/D2Common/src/Items/Items.cpp

```diff
@@ -1004,32 +1004,51 @@ void __stdcall ITEMS_SetOrRemoveFlag(int* pFlags, int nFlag, BOOL bSet)
 	}
 }
 
-//D2Common.0x6FD99740 (#10756)
-BOOL __stdcall ITEMS_CheckRequirements(D2UnitStrc* pItem, D2UnitStrc* pUnit, BOOL bEquipping, BOOL* bStrength, BOOL* bDexterity, BOOL* bLevel)
+static int ITEMS_GetBonusStatFromSockets(D2UnitStrc* pItem, D2C_ItemStats stat)
 {
-	D2ItemTypesTxt* pItemTypesTxtRecord = NULL;
-	D2ItemsTxt* pItemsTxtRecord = NULL;
-	D2ItemDataStrc* pItemData = NULL;
+	int nStatBonusFromSockets = 0;
+	if (pItem->dwUnitType == UNIT_ITEM)
+	{
+		if (ITEMS_CheckIfSocketableByItemId(pItem->dwClassId))
+		{
+			for (D2UnitStrc* i = INVENTORY_GetFirstItem(pItem->pInventory); i != NULL; i = INVENTORY_GetNextItem(i))
+			{
+				i = INVENTORY_UnitIsItem(i);
+				if (pItem->dwAnimMode == IMODE_EQUIP)
+				{
+					nStatBonusFromSockets += STATLIST_GetUnitStat(i, stat, 0);
+				}
+			}
+		}
+	}
+	return nStatBonusFromSockets;
+}
 
-	BOOL bIdentified = FALSE;
-	BOOL bStrReqMet = FALSE;
-	BOOL bDexReqMet = FALSE;
-	BOOL bLevelReqMet = FALSE;
+static bool ITEMS_CheckStatRequirement(D2UnitStrc* pItem, D2UnitStrc* pUnit, D2C_ItemStats stat, BOOL bEquipping, int nStatWithPctBonus)
+{
+
+	BOOL bStatReqMet = FALSE;
+	const int nUnitStat = STATLIST_GetUnitStat(pUnit, stat, 0);
+	if (nUnitStat > 0 && nUnitStat >= nStatWithPctBonus)
+	{
+		bStatReqMet = TRUE;
+
+		if (bEquipping && STATLIST_GetOwner(pItem, 0))
```

Comment:
Can you add a comment here, that STATLIST_GetOwner was not checked for dexterity in the original game?

Reply:
Sure!